File: ruby/hamming/hamming.rb
class Hamming
  def Hamming.compute(s1, s2)
    if s1.size > s2.size
      Hamming.compute(s2, s1)
    else
      s1.chars.zip(s2.chars).count{|c1, c2| c1 != c2 }
    end
  end
end

File: exercism/javascript/hamming/hamming.js
var Hamming = function() { };

Hamming.prototype = {
	compute: function(s1,s2) {
		if(s1.length != s2.length) throw new Error("DNA strands must be of equal length.");
		var diffCount = 0;
		var _g1 = 0;
		var _g = s1.length;
		while(_g1 < _g) {
			var i = _g1++;
			if(s1.charAt(i) != s2.charAt(i)) diffCount++;
		}
		return diffCount;
	}
};

module.exports = Hamming;
